### PR TITLE
Add authority to account

### PR DESCRIPTION
**Pull request by shahzaibj**
Allows developers/users to call `getAuthority` on Account object so they can get that authority and use it for making silent calls.

Fixed for the case of AAD by just grabbing from the issuer claim inside the ID Token. The authority return by `getAuthority` for the case of B2C will be wrong, and that will be fixed in a future PR.

### CODE REVIEW

File: msal/src/main/java/com/microsoft/identity/client/Account.java [hunk 1]
```diff
@@ -146,4 +147,17 @@ public String getUsername() {
 
         return MISSING_FROM_THE_TOKEN_RESPONSE;
     }
+
+    @Override
+    @NonNull
+    public String getAuthority() {
+        if (null != getClaims()) {
```
Comment (rpdome) [inline msal/src/main/java/com/microsoft/identity/client/Account.java]: nit: please add TODO here.

File: msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/AcquireTokenNetworkB2CTest.java [hunk 2]
```diff
@@ -33,6 +34,11 @@
  */
 public class AcquireTokenNetworkB2CTest extends AcquireTokenNetworkTest {
 
+    @Override
+    public String getAuthority() {
+        return mApplication.getConfiguration().getDefaultAuthority().getAuthorityURL().toString();
```
Comment (rpdome) [inline msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/AcquireTokenNetworkB2CTest.java]: nit: also add TODO here.

File: msal/src/main/java/com/microsoft/identity/client/Account.java (hunk 1 above)
Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/Account.java]: Why?

Are you referring to the case of B2C?

Comment (rpdome) [reply msal/src/main/java/com/microsoft/identity/client/Account.java]: yes. we should add TODO for any in-progress/partially finished work.

Comment (shahzaibj) [reply msal/src/main/java/com/microsoft/identity/client/Account.java]: Addressed

File: msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/AcquireTokenNetworkB2CTest.java (hunk 2 above)
Comment (shahzaibj) [reply msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/AcquireTokenNetworkB2CTest.java]: Addressed

File: msal/src/test/java/com/microsoft/identity/client/e2e/shadows/ShadowAuthority.java [hunk 3]
```diff
@@ -99,7 +100,8 @@ public static Authority getAuthorityFromAuthorityUrl(String authorityUrl) {
             case AAD_MOCK_DELAYED_PATH_SEGMENT:
                 authority = new MockDelayedResponseAuthority();
                 break;
-            case B2C_TEST_PATH_SEGMENT:
+            case B2C_PATH_SEGMENT:
+            case B2C_PATH_SEGMENT_ALT:
```
Comment (iambmelt) [inline msal/src/test/java/com/microsoft/identity/client/e2e/shadows/ShadowAuthority.java]: 💪

**Pull request merged**